src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -211,4 +259,47 @@ public String sanitizeInput(String input) throws java.io.UnsupportedEncodingExce
       input = new String( input.getBytes("UTF-8") , 0, Math.min(MAX_STRING_BYTES, input.length()), "UTF-8");
       return input;
   }
+
+  public long getEventId() {
+      Query query = new Query(EVENT_NUM);
+      PreparedQuery pq = datastore.prepare(query);
+      QueryResultList<Entity> eventNumResult = pq.asQueryResultList(FetchOptions.Builder.withDefaults());
+
+      Entity eventNumEntity;
+      long eventNumValue;
+      if (eventNumResult.size() == 0) {
+          eventNumEntity = new Entity(EVENT_NUM);
+          eventNumValue = 0;
+          eventNumEntity.setProperty(EVENT_NUM_VALUE, eventNumValue);
+          datastore.put(eventNumEntity);      
+      }
+      else {
+          eventNumEntity = eventNumResult.get(0);
+          eventNumValue = (long)eventNumEntity.getProperty(EVENT_NUM_VALUE)+1;
+          eventNumEntity.setProperty(EVENT_NUM_VALUE, eventNumValue);
+          datastore.put(eventNumEntity);
+      }
+
+      return eventNumValue;
+  }
+
+  public void updateUserCreatedEvents(String userId, long eventId) {
+      Query userQuery = new Query(User.DATA_TYPE).setFilter(new FilterPredicate(User.ID, FilterOperator.EQUAL, userId));
+      Entity entity = datastore.prepare(userQuery).asSingleEntity();
+      ArrayList<Long> createdEvents =(ArrayList<Long>) entity.getProperty(User.CREATED_EVENTS);
+      if (createdEvents == null) createdEvents = new ArrayList<Long>();
+      createdEvents.add(eventId);
+      entity.setProperty(User.CREATED_EVENTS, createdEvents);
+      datastore.put(entity);
```

Comment:
Not fully familiar with datastore. Is there a way to mention the key for a datastore? I seeing that you are inserting a value (entity) in this case. Is there a way to define a PRIMARY KEY for a datastore table?

Reply:
There's a way to set the id which is what I am doing with the entity_key in the User class, but I'm not sure about making another attribute the primary key